### PR TITLE
Feature/implement graphql filters

### PR DESCRIPTION
**Pull request by dairbuirabass**
## Description
Add support for Magento GraphQl filters.
Implement mapping between Nosto and Magento filtering (e.g. color, brand, category).

#### Custom fields filtering ✅ 
1. Yes/No
2. Multi-select 
3. Dropdown
4. Text Swatch
5. Visual Swatch

#### Categories ❌ 
To be done (not possible to send complex category query for layered navigation).

#### Ratings ❌ 
Not possible to filter through ratings in Magento 2 GraphQl.

## Motivation and Context
Support PWA implementations (PWA studio, Venia)

## How Has This Been Tested?
Postman, PHP 8.1, Magento 2.4.4

## Documentation:
TO DO: Update https://docs.nosto.com/magento-2/addons/cmp/guides/pwa-implementation#filters

## Checklist:
- [x] My code follows the code style of this project.
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] I have assigned the correct milestone or created one if non-existent.
- [x] I have correctly labeled this pull request.
- [x] I have linked the corresponding issue in this description.
- [x] I have updated the corresponding Jira ticket.
- [x] I have requested a review from at least 2 reviewers
- [x] I have checked the base branch of this pull request
- [x] I have checked my code for any possible security vulnerabilities

### CODE REVIEW

File: Model/Service/Facet/BuildGraphQlFacetService.php [hunk 1]
```diff
@@ -52,12 +84,49 @@ public function getFacets(array $requestData): Facet
...
+                        foreach ($filterValues as $value) {
+                            /** @phan-suppress-next-next-line PhanUndeclaredMethod */
+                            /** @noinspection PhpUndefinedMethodInspection */
+                            $values[] = $attribute->getSource()->getOptionText($value);
+                        }
+                    }
+
+                    $brandAttribute = $this->nostoDataHelper->getBrandAttribute();
+
+                    if ($attributeCode === $brandAttribute) {
```
Comment (supercid) [inline Model/Service/Facet/BuildGraphQlFacetService.php:118]: might not even need the `??` to null
```suggestion
                    if ($attributeCode === $this->nostoDataHelper->getBrandAttribute() ?? null) {
```

File: Model/Service/Facet/BuildGraphQlFacetService.php [hunk 2]
```diff
@@ -52,12 +84,49 @@ public function getFacets(array $requestData): Facet
...
+                    isset($filter['to']) ? $filter['to'] : null
+                );
+            } else { // Custom field filters
+                try {
+                    $attributeCode = $filter['field'];
+                    $filterValues = $filter['value'];
+                    $attribute = $this->productAttributeRepository->get($attributeCode);
+                    $values = [];
+
+                    if (is_string($filterValues)) { // eq attribute
```
Comment (supercid) [inline Model/Service/Facet/BuildGraphQlFacetService.php:104]: I'd also check for `''`, will `getOptionText` crash or return empty?

**Review by supercid — COMMENTED** [review]
looks good, but I'd try to split this into smaller methods

File: Model/Service/Facet/BuildGraphQlFacetService.php (hunk 2 above)
Comment (dairbuirabass) [reply Model/Service/Facet/BuildGraphQlFacetService.php:114]: Changed the whole logic in the new method. Also, filters are not even generated if the request is empty string.

**Review by supercid — APPROVED** [review]

**Pull request merged**